README.md

```diff
@@ -498,3 +498,51 @@ Some of the options include:
 * `members` - automatically document the members in that Python module.
 * `private-members` - include private functions and methods.
 * `show-inheritance` - show the class inheritance.
+
+# Opentelemetry Instrumentation
+
+IIB is now implemented with Opentelemetry python instrumentation.
+This is done in order to collect detailed information about the behavior and performance of IIB.
+By using a combination of automatic and manual instrumentation, it is possible to capture all relevant information about the application's behavior.
+
+## Environment setup
+The two most important packages to be installed are the `opentelemetry-api` and `opentelemetry-sdk`.
+Please see the ``requirements.txt`` file for a more detailed list..
+
+```bash
+pip install virtualenv
+virtualenv venv
+source venv/bin/activate
+pip install -r requirements.txt
+```
+
+## Traces
+The application is instrumented to collect traces using OpenTelemetry.
+The traces include information about the duration of each request, as well as any errors that may have occurred.
+
+## Opentelemetry Environment Variables
+For your local deployment to use OpenTelemetry, the following variables need to be defined in `docker-compose.yml` and `podman-compose.yml`.
+
+* `OTEL_EXPORTER_OTLP_ENDPOINT` - The endpoint for the OpenTelemetry exporter.
+* `OTEL_SERVICE_NAME` - "iib-workers"
+* `OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST` - captures headers for the incoming requests
+* `OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE` - captures headers for the outgoing requests
+* `OTEL_PROPAGATORS` - specifies the propagators that should be used by the OpenTelemetry tracer to propagate context across distributed systems.
+* `OTEL_EXPORTER_OTLP_TRACES_PROTOCOL` This to configure the protocol used by the OTLP exporter to send traces to a backend. We need to set it as "http/protobuf"`
```

Comment:
Could you please point me to where these are used in the code as I don't see them?